component/component/vshn_appcat_services.jsonnet

```diff
@@ -0,0 +1,213 @@
+local com = import 'lib/commodore.libjsonnet';
+local kap = import 'lib/kapitan.libjsonnet';
+local kube = import 'lib/kube.libjsonnet';
+
+local comp = import 'lib/appcat-compositions.libsonnet';
+local crossplane = import 'lib/crossplane.libsonnet';
+
+local common = import 'common.libsonnet';
+local prom = import 'prometheus.libsonnet';
+local xrds = import 'xrds.libsonnet';
+
+local slos = import 'slos.libsonnet';
+
+local inv = kap.inventory();
+local params = inv.parameters.appcat;
+
+local serviceNameLabelKey = 'appcat.vshn.io/servicename';
+local serviceNamespaceLabelKey = 'appcat.vshn.io/claim-namespace';
+
+local getServiceNamePlural(serviceName) =
+  local serviceNameLower = std.asciiLower(serviceName);
+  if std.endsWith(serviceName, 's') then
+    serviceNameLower
+  else
+    serviceNameLower + 's';
+
+local vshn_appcat_service(name) =
+  local isOpenshift = std.startsWith(inv.parameters.facts.distribution, 'openshift');
+  local securityContext = if isOpenshift then false else true;
+
+  local serviceParams = params.services.vshn.services[name];
+  local connectionSecretKeys = serviceParams.connectionSecretKeys;
+  local promRuleSLA = prom.PromRuleSLA(serviceParams.sla, serviceParams.serviceName);
+  local plans = common.FilterDisabledParams(serviceParams.plans);
+  local serviceNamePlural = getServiceNamePlural(serviceParams.serviceName);
+
+
+  local restoreServiceAccount = kube.ServiceAccount(serviceParams.restoreSA) + {
+    metadata+: {
+      namespace: params.services.controlNamespace,
+    },
+  };
+
+  local restoreRoleName = 'crossplane:appcat:job:' + name + ':restorejob';
+  local restoreRole = kube.ClusterRole(restoreRoleName) {
+    rules: serviceParams.restoreRoleRules,
+  };
+
+  local restoreClusterRoleBinding = kube.ClusterRoleBinding('appcat:job:' + name + ':restorejob') + {
+    roleRef_: restoreRole,
+    subjects_: [ restoreServiceAccount ],
+  };
+
+  local resizeServiceAccount = kube.ServiceAccount('sa-sts-deleter') + {
+    metadata+: {
+      namespace: params.services.controlNamespace,
+    },
+  };
+
+  local resizeClusterRole = kube.ClusterRole('appcat:job:' + name + ':resizejob') {
```

Comment:
Wouldn't it make more sense to just have one set of clusterRoles and service accounts for all helm based services? The way this is currently implemented they all get the same hardcoded clusterRoles anyway.

Reply:
Removed it from this file as well as the redis file (to avoid duplicate manifests) and moved it to the sts-resize-controller file.